api/v1alpha3/types.go

```diff
@@ -179,7 +179,14 @@ type NetworkSpec struct {
 	// Subnets configuration.
 	// +optional
 	Subnets Subnets `json:"subnets,omitempty"`
-}
+
+	// TagSecurityGroups optionally disables tagging the security groups used by the cluster
+	// +optional
+	TagSecurityGroups *bool `json:"tagSecurityGroups,omitempty"`
+
+	// SecurityGroupOverrides is an optional set of security groups to use for cluster instances
+	// +optional
+	SecurityGroupOverrides map[SecurityGroupRole]AWSResourceReference `json:"securityGroupOverrides,omitempty"`}
```

Comment:
Do we actually expect users to provide more information here than possibly a SecurityGroup ID? If not, then this should probably use a new type, otherwise users might be confused by what options would be respected/ignored when used in this context.

This has already caused confusion in the existing usage of the AWSResourceReference type.

Reply:
Cool, I'll make this `map[SecurityGroupRole]string`

Reply:
sgtm, thanks!